pygccxml/declarations/type_traits.py

```diff
@@ -284,19 +284,46 @@ def remove_reference(type):
 def is_const(type):
     """returns True, if type represents C++ const type, False otherwise"""
     nake_type = remove_alias(type)
-    return isinstance(nake_type, cpptypes.const_t)
+    if isinstance(nake_type, cpptypes.const_t):
+        return True
+    elif isinstance(nake_type, cpptypes.volatile_t):
+        return is_const(nake_type.base)
+    elif isinstance(nake_type, cpptypes.array_t):
+        return is_const(nake_type.base)
+    return False
 
 
-def remove_const(type):
+def remove_const(type_):
     """removes const from the type definition
 
     If type is not const type, it will be returned as is
     """
 
-    nake_type = remove_alias(type)
+    nake_type = remove_alias(type_)
     if not is_const(nake_type):
-        return type
+        return type_
     else:
+        # Handling for const and volatile qualified types. There is a
+        # difference in behavior between GCCXML and CastXML for cv-qual arrays.
+        # GCCXML produces the following nesting of types:
+        #       ->  volatile_t(const_t(array_t))
+        # while CastXML produces the following nesting:
+        #       ->  array_t(volatile_t(const_t))
+        # For both cases, we must unwrap the types, remove const_t, and add
+        # back the outer layers
+        if isinstance(nake_type, cpptypes.array_t):
+            is_v = is_volatile(nake_type)
+            if is_v:
+                result_type = nake_type.base.base.base
+            else:
+                result_type = nake_type.base.base
+            if is_v:
+                result_type = cpptypes.volatile_t(result_type)
+            return cpptypes.array_t(result_type, nake_type.size)
+
+        elif isinstance(nake_type, cpptypes.volatile_t):
```

Comment:
`remove_const` wasn't fully fixed by the previous commits. The code I added above took care of `T const volatile[N]` types, but simple `T const volatile` types were still not being handled correctly. This `elif` takes care of that case.